data-prepper-plugins/dynamodb-source/src/main/java/org/opensearch/dataprepper/plugins/source/dynamodb/stream/ShardConsumer.java

```diff
@@ -230,10 +243,15 @@ public void run() {
                     records = response.records();
                 }
                 recordConverter.writeToBuffer(acknowledgementSet, records);
+                long delay = System.currentTimeMillis() - lastEventTime.toEpochMilli();
+                interval = delay > GET_RECORD_DELAY_THRESHOLD_MILLS ? MINIMUM_GET_RECORD_INTERVAL_MILLS : GET_RECORD_INTERVAL_MILLS;
```

Comment:
Can you explain the theory behind this a bit? Why reduce interval to the minimum when the delay is greater (meaning traffic is less)?

Reply:
Need to make more calls per second to `catch up`.

Reply:
For example, if each call takes 200ms, and right now, it can only make roughly 2 calls per second. But with this change, it can call up to 4 or 5 calls per second to get to the latest point.

Reply:
Oh I get it. I was thinking this delay was the wait time with no records for some reason. Thanks for explaining.

Reply:
Is the idea here that we don't want to make too many calls when the last event is near to time in order to avoid unnecessary congestion with DynamoDB? But, when we notice that events are getting behind, we make more calls?